<launch>

   <master auto="start"/>

 <!-- Run Gmapping --> 
   <!-- <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
        <param name="base_frame" value="base_link"/>
        <param name="odom_frame" value="odom"/>
        <param name="map_frame" value="map"/>
        <param name="maxUrange" value="6.0"/>
        <param name="maxRange" value="8.0"/>
        <param name="sigma" value="0.05"/>
        <param name="kernelSize" value="1"/>
        <param name="lstep" value="0.05"/>
        <param name="astep" value="0.05"/>
        <param name="iterations" value="5"/>
        <param name="lsigma" value="0.075"/>
        <param name="ogain" value="3.0"/>
        <param name="lskip" value="0"/>
        <param name="srr" value="0.1"/>
        <param name="srt" value="0.2"/>
        <param name="str" value="0.1"/>
        <param name="stt" value="0.2"/>
        <param name="linearUpdate" value="0.1"/>
        <param name="angularUpdate" value="0.1"/>
        <param name="temporalUpdate" value="-1.0"/>
        <param name="resampleThreshold" value="0.5"/>
        <param name="particles" value="30"/>
        <param name="xmin" value="-50.0"/>
        <param name="ymin" value="-50.0"/>
        <param name="xmax" value="50.0"/>
        <param name="ymax" value="50.0"/>
        <param name="delta" value="0.05"/>
        <param name="llsamplerange" value="0.01"/>
        <param name="llsamplestep" value="0.01"/>
        <param name="lasamplerange" value="0.005"/>
        <param name="lasamplestep" value="0.005"/>
        <param name="tf_delay" value="0.3"/>
        <param name="map_update_interval" value="5.0"/>
        <param name="occ_thresh" value="0.25"/>
        <param name="laser_z" value="0.54"/>
        <param name="angularUpdate" value="0.5"/>
        <param name="linearUpdate" value="0.2"/>
    </node> -->

<!-- Run the map server --> 
    <!-- <node name="map_server" pkg="map_server" type="map_server" args="$(find followme)/world/Hallway/hallway.yaml"/> -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find followme)/world/IS3L_map/isr-150524.yaml"/>
  

<!-- Run AMCL --> 
    <include file="$(find followme)/launch/amcl.launch" /> 

<!-- Run Move Base -->
   <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <!-- <param name="base_global_planner" value="carrot_planner/CarrotPlanner"/>  -->

    <!-- Disable recovery behaviors -->
        <rosparam param="recovery_behaviors">
            []
        </rosparam>

    <param name="base_local_planner" value="followperson_local_planner/FollowPersonLocalPlanner" />
    <rosparam file="$(find followme)/params/followperson_local_planner.yaml" command="load" />
    <rosparam file="$(find followme)/params/costmap_common_params.yaml" command="load" ns="global_costmap" /> 
    <rosparam file="$(find followme)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find followme)/params/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find followme)/params/global_costmap_params.yaml" command="load" /> 
    <rosparam file="$(find followme)/params/base_local_planner_params.yaml" command="load" />  

    <!-- Load custom plugin parameters -->
    <rosparam file="$(find followme)/params/plugins_params.yaml" command="load" />
   </node>

</launch> 